Zend/zend_object_handlers.c

```diff
@@ -164,27 +164,30 @@ ZEND_API HashTable *zend_std_get_debug_info(zend_object *object, int *is_temp) /
 	}
 
 	zend_call_known_instance_method_with_0_params(ce->__debugInfo, object, &retval);
-	if (Z_TYPE(retval) == IS_ARRAY) {
-		if (!Z_REFCOUNTED(retval)) {
-			*is_temp = 1;
-			return zend_array_dup(Z_ARRVAL(retval));
-		} else if (Z_REFCOUNT(retval) <= 1) {
+	switch (Z_TYPE(retval)) {
+		case IS_ARRAY:
+			if (!Z_REFCOUNTED(retval)) {
+				*is_temp = 1;
+				return zend_array_dup(Z_ARRVAL(retval));
+			} else if (Z_REFCOUNT(retval) <= 1) {
+				*is_temp = 1;
+				ht = Z_ARR(retval);
+				return ht;
+			} else {
+				*is_temp = 0;
+				zval_ptr_dtor(&retval);
+				return Z_ARRVAL(retval);
+			}
+		default:
+			zend_throw_error(NULL, ZEND_DEBUGINFO_FUNC_NAME "() must return an array");
```

Comment:
this code should not be reached in case of an exception has been thrown

also, the `print_r`/`var_dump` statement should not print anything when an exception has been raised